Sources/LiveViewNative/Modifiers/Toolbars Modifiers/ToolbarModifier.swift

```diff
@@ -0,0 +1,248 @@
+///         <ToolbarItem id="search" placement="secondary-action">
+///             <TextField text-field-style="rounded-border">Search for photos...</TextField>
+///         </ToolbarItem>
+///         <ToolbarItem id="actions" customization-behavior="disabled">
+///             <Button phx-click="search">Search</Button>
+///         </ToolbarItem>
+///         <ToolbarItem id="history" visibility="hidden" always-available>
+///             <Button phx-click="search">Open History</Button>
+///         </ToolbarItem>
+///         <ToolbarItem id="spacer" visibility="hidden">
+///             <Spacer />
+///         </ToolbarItem>
+///     </toolbar:editable-bar>
+/// </Text>
+/// ```
+///
+/// ## Arguments
+/// * ``id``
+/// * ``content``
+///
+/// ## See Also
+/// ### Toolbars
+/// * ``ToolbarItem``
+/// * ``ToolbarItemGroup``
+/// * ``ToolbarTitleMenu``
+#if swift(>=5.8)
+@_documentation(visibility: public)
+#endif
+struct ToolbarModifier<R: RootRegistry>: ViewModifier, Decodable {
+    /// A unique identifier for a customizable toolbar.
+    #if swift(>=5.8)
+    @_documentation(visibility: public)
+    #endif
+    private let id: String?
+
+    /// The name of the element with the toolbar content.
+    #if swift(>=5.8)
+    @_documentation(visibility: public)
+    #endif
+    private let content: String
+    
+    @ObservedElement private var element
+    @LiveContext<R> private var context
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        self.id = try container.decodeIfPresent(String.self, forKey: .id)
+        self.content = try container.decode(String.self, forKey: .content)
+    }
+
+    func body(content: Content) -> some View {
+        if let id {
+            content.toolbar(id: id) {
+                CustomizableToolbarTreeBuilder<R>().fromNodes(
+                    context.children(of: element, withTagName: self.content, namespace: "toolbar").flatMap { $0.children() },
+                    context: context.storage
+                )
+            }
+        } else {
+            content.toolbar {
+                ToolbarTreeBuilder<R>().fromNodes(
+                    context.children(of: element, withTagName: self.content, namespace: "toolbar").flatMap { $0.children() },
+                    context: context.storage
+                )
+            }
+        }
+    }
+
+    enum CodingKeys: String, CodingKey {
+        case id
+        case content
+    }
+}
+
+/// A builder for `ToolbarContent`.
+struct ToolbarTreeBuilder<R: RootRegistry> {
+    func fromNodes<Nodes>(_ e: Nodes, context c: LiveContextStorage<R>) -> some ToolbarContent
+        where Nodes: RandomAccessCollection, Nodes.Index == Int, Nodes.Element == Node
+    {
+        switch e.count {
+        case 1:
+            return ToolbarContentBuilder.buildBlock(f(e[0], c), f(nil, c), f(nil, c), f(nil, c), f(nil, c), f(nil, c), f(nil, c), f(nil, c), f(nil, c), f(nil, c))
```

Comment:
This was necessary for the compiler to type check in a reasonable time. Unfortunately the `TupleToolbarContent` type is internal unlike `TupleView`.